spec/dispatcher.spec.js

```diff
@@ -242,34 +242,23 @@ describe('Dispatcher', function () {
       it('fires newline when shift + enter is pressed', (done) => {
         on('newline', () => done())
         shiftReturn(elem)
-        expect(elem.innerHTML).to.equal('<br>\u0000')
+        expect(elem.innerHTML).to.equal('<br>\uFEFF')
       })
 
-      it('creates a nested br element with a data-editable="unwrap" attribute', () => {
+      it('appends a zero-width space after the br tag to force a line break', () => {
         typeKeys(elem, 'foobar')
         shiftReturn(elem)
         expect(elem.innerHTML).to.equal(
-          `\u0000` +
-          `foobar` +
-          `<span data-editable="unwrap">` +
-            `<br><span data-editable="remove" contenteditable="false">\u0000</span>` +
-          `</span>`
+          `\uFEFFfoobar<br>\uFEFF`
         )
       })
 
-      it('removes the nested br element when adding a newline afterwards', () => {
+      it('does not append another zero-width space when one is present already', () => {
         typeKeys(elem, 'foobar')
         shiftReturn(elem)
         shiftReturn(elem)
         expect(elem.innerHTML).to.equal(
-          `\u0000` +
-          `foobar` +
-          // this is the nested br element that got unwrapped
-          `<br>` +
-          // This is the new nested newline of the second shift + return
-          `<span data-editable="unwrap">` +
-            `<br><span data-editable="remove" contenteditable="false">\u0000</span>` +
-          `</span>`
+          `\uFEFFfoobar<br><br>\uFEFF`
```

Comment:
I've tried to write a test for the deletion, but it's not really possible to trigger a backspace or `InputEvent` of type `deleteContentBackward` that mutates the content using javascript 😢.
We might need to simulate it using manual operations, which basically means reimplementing the backspace behavior.